src/main/java/org/terasology/scenario/internal/ui/HubToolScreen.java

```diff
@@ -452,6 +453,27 @@ public void onOpened() {
         }
     }
 
+
+    @Override
+    public void update(float delta) {

```

Comment:
This requires an explanation why a dirty flag gets set and why not the code of the update method gets executed right when it becomes dirty. Maybe the explanation is somewhere else and I missed, then a reference will be good. Possibly you want to extract the code of the update method into it's own method and add the explanation to the javadoc of that method.